src/tests/snapshots/mast.rs

```diff
@@ -0,0 +1,305 @@
+use crate::negative_tests::mast_pass;
+
+fn test_monomorphization(code_snippet: &str) {
+    let mast = mast_pass(code_snippet).expect("Mast pass failed!");
+
+    // Test functions
+    for (fully_qualified, fn_info) in &mast.0.functions {
+        insta::assert_snapshot!(serde_json::to_string_pretty(fully_qualified).unwrap());
+        insta::assert_snapshot!(serde_json::to_string_pretty(fn_info).unwrap());
+    }
+
+    // Test node types
+    for (counter, ty_kind) in &mast.0.node_types {
+        insta::assert_snapshot!(serde_json::to_string_pretty(counter).unwrap());
+        insta::assert_snapshot!(serde_json::to_string_pretty(ty_kind).unwrap());
+    }
+
+    // Test structs/constants/node_id
+    insta::assert_snapshot!(serde_json::to_string_pretty(&mast.0.structs).unwrap());
+    insta::assert_snapshot!(serde_json::to_string_pretty(&mast.0.constants).unwrap());
+    insta::assert_snapshot!(serde_json::to_string_pretty(&mast.0.node_id).unwrap());
+}
+
+#[test]
+fn test_generic_array_access() {
+    let code = r#"
+    fn last(arr: [Field; LEN]) -> Field {
+        return arr[LEN - 1];
+    }
+    fn main(pub xx: Field) {
+        let array1 = [xx + 1, xx + 2, xx + 3];
+        let elm1 = last(array1);
+        assert_eq(elm1, 4);
+    }
+    "#;
+    test_monomorphization(code);
+}
+
+#[test]
+fn test_generic_array_nested() {
+    let code = r#"
+    fn last(arr: [[Field; NN]; 3]) -> Field {
+        let mut newarr = [0; NN * 3];
+    
+        let mut index = 0;
+        for ii in 0..3 {
+            let inner = arr[ii];
+            for jj in 0..NN {
+                newarr[(ii * NN) + jj] = inner[jj];
+            }
+        }
+    
+        return newarr[(NN * 3) - 1];
+    }
+    
+    fn main(pub xx: Field) {
+        let array = [[xx + 1, xx + 2, xx + 3]; 3];
+        let elm = last(array);
+        assert_eq(elm, 4);
+    }
+    "#;
+    test_monomorphization(code);
+}
+
+#[test]
+fn test_generic_builtin_bits() {
```

Comment:
This tests fails [here](https://github.com/cyberbono3/noname/blob/test-mast/src/mast/mod.rs#L596)

Reply:
Basically, I have performed snapshot testing for all fields of `Mast(pub TypeChecker). `